### PR TITLE
Nuke op toys can now actually explode

### PR DESCRIPTION
**Pull request by ghost**
Emagging the nuke op toy make it able to explode after five seconds and give a warning right before exploding, does not breach

image

:cl:
 * rscadd: Nuke op toys can be emagged to turn it into a bomb which explodes after a few seconds but gives a warning right before exploding

### CODE REVIEW

**Comment by Megamaw** [comment]
great, now add it to the beer nuke

edit: right, poe's law. i'm mostly joking.

**Comment by ghost** [comment]
I hope I didn't fuck up the spacing or indenting

**Comment by nuklearcellphoneg** [comment]
Only if it explodes instantly in hand.

**Comment by ghost** [comment]
So emag then active in hand to explode?

**Comment by ghost** [comment]
Getting the specific toy is kinda hard

**Comment by nuklearcellphoneg** [comment]
Well, it would make more sense, in line with how Nuke Ops *deathgasp

In fact, making the toy normally have a *deathgasp line, like it's a cool toy and and using it in hand makes it 'vibrates violently, its eyes lighting up, before falling still, its eyes dim and lightless', and then when emagged it does the same thing, except after this it explodes would be even better.

**Comment by ghost** [comment]
I think there's another variant of toys where they actually say things related to what they mimic.
There's gasha and action figures, I'll probably add the deathgasp message when it's emagged

**Comment by ComicIronic** [comment]
Can't you clickspam this for the explosion noise a lot before it actually explodes?

**Comment by ghost** [comment]
Would it still make noise if it's qdeleted

**Comment by Exxion** [comment]
Technically no, but sort of
It would shit up the runtime log though

**Comment by Riekland** [comment]
That explosion size looks a little big to me
Is that the same size as an explosive implant?

**Comment by ghost** [comment]
Explosive implant is 1,3,4,6
This one is 1,2,4
Not sure what the last number 6 is

**Comment by Exxion** [comment]
It's supposed to be flash range, but it doesn't actually do anything

**Comment by yclatious** [comment]
Why would a toy, when emagged, explode like a 1tank or explosive implant? I get it, its a joke on ops, but this seems TG tier to me and a free 1dev that a miner could just shit out if he made some coins or something and wasted time, essentially just free 1tanks, yeah its hypothetical but doesn't change the fact its a really shitty joke that uses the emag as a excuse to exist, its a toy, why would it explode and breach a space station, its just a toy.

**Comment by Riekland** [comment]
Yeah I'd make it a lot less powerful
Maybe even just not making it breach areas would be fine for me, free breaches are gay

**Comment by ghost** [comment]
>free breaches are gay
There are a lot of free breaches in maint.
I GUESS I can make it not breach

**Comment by Blithering** [comment]
Replace the click with "Someone pass the boombox"

**Comment by yclatious** [comment]
Free breaches are indeed gay, the free breaches in maint make sense and a toy that explodes wouldnt even have enough force to brach anyhow.

**Comment by Durgen** [comment]
maybe give it X% chance to blow up instantly when emagged so it doesn't work every time

**Comment by TheFriggenSawbones** [comment]
i like the idea but i still feel like the explosions too big

**Comment by ghost** [comment]
explosion is still big but doesn't breach and now it counts down from when you emag it

oh and it clicks at emagging then says "pass the boombox" right before exploding

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -760,6 +760,24 @@
...
+
+/obj/item/toy/gasha/newcop/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/weapon/card/emag) && !emagged)
+		to_chat(user, "<span class='warning'>You turned the toy into a bomb!</span>")
+		emagged = 1
+
+		playsound(get_turf(src), 'sound/effects/kirakrik.ogg', 100, 1)
+
+		sleep(50)
+		src.say("Someone pass the boombox")
```
Comment (Intigracy) [inline code/game/objects/items/toys.dm]: add a .
remove the src. part of it, it's unnecessary

File: code/game/objects/items/toys.dm [hunk 2]
```diff
@@ -760,6 +760,24 @@
 	name = "toy nuke-op"
 	desc = "Mildly explosive."
 	icon_state = "newcop"
+	var/emagged = 0
+
+/obj/item/toy/gasha/newcop/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/weapon/card/emag) && !emagged)
```
Comment (Intigracy) [inline code/game/objects/items/toys.dm]: if(isEmag(I) && !emagged)

**Review by Intigracy — DISMISSED** [review]

**Comment by ghost** [comment]
>gameplay/gamemode

???

**Comment by ghost** [comment]
madman

**Pull request merged**